app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala

```diff
@@ -160,4 +169,27 @@ class NotificationMongoRepository @Inject()(mongoDbProvider: MongoDbProvider,
     collection.find(selector, Some(projection)).cursor[NotificationWithIdOnly]().collect[List](Int.MaxValue, Cursor.FailOnError[List[NotificationWithIdOnly]]())
   }
 
+  override def fetchNotificationIds(clientId: String, conversationId: UUID, notificationStatus: NotificationStatus.Value): Future[List[NotificationWithIdOnly]] = {
+    val selector = notificationStatus match {
+      case Pulled => Json.obj("clientId" -> clientId, "notification.headers.X-Conversation-ID" -> conversationId, "notification.datePulled" -> Json.obj("$exists" -> true))
+      case Unpulled => Json.obj("clientId" -> clientId, "notification.datePulled" -> Json.obj("$exists" -> false))
```

Comment:
`notification.headers.X-Conversation-ID` for unpulled?

Reply:
Fixed